src/server/acl/user.cc

```diff
@@ -66,22 +74,47 @@ bool User::HasPassword(std::string_view password) const {
   return *password_hash_ == StringSHA256(password);
 }
 
-void User::SetAclCategories(uint64_t cat) {
+void User::SetAclCategories(uint32_t cat) {
   acl_categories_ |= cat;
 }
 
-void User::UnsetAclCategories(uint64_t cat) {
+void User::UnsetAclCategories(uint32_t cat) {
   SetAclCategories(cat);
   acl_categories_ ^= cat;
 }
 
+void User::SetAclCommands(size_t index, uint64_t bit_index, bool all) {
+  if (all) {
+    for (auto& family : commands_) {
+      family = ALL_COMMANDS;
+    }
+    return;
+  }
+  commands_[index] |= bit_index;
+}
```

Comment:
its kinda confusing that with the all argument your index & bit_index have no meaning

Reply:
I agree on this, let me see what I can do

Reply:
I ended up using the index, if it's numeric_limits::max() then it denotes an `all`. I also added a comment